src/syncManager.ts

```diff
@@ -72,16 +73,16 @@ export class SyncManager {
 
             if (batchArray.length === this.tilesConfig.uploadBatchSize) {
               await Promise.all(batchArray);
-              await this.tilesManager.updateTilesCount(layerId, batchArray.length);
+              uploadedTiles += batchArray.length;
               batchArray = [];
             }
           }
           // resolved left overs
           await Promise.all(batchArray);
-
-          await this.tilesManager.updateTilesCount(layerId, batchArray.length);
+          uploadedTiles += batchArray.length;
 
           await this.queueClient.queueHandler.ack(jobId, taskId);
+          await this.tilesManager.updateTilesCount(layerId, uploadedTiles);
```

Comment:
if the queue ack pass but the update count fail it will result in sync with potential missing tiles or tiles that change after the layer is published that are hard to track.
if you do it the other way around and the ack fail it will result in never ending sync that is easier to find. also the ack is less likely to fail because the job used the queue manager to retrieve the task so its less likely to be down when that line is reached.

the best solution would be calling the increment first and then the ack and trying to role back the count if the ack fail.

ill approve it because the error chance are low but take this into consideration before merging